File: src/client/java/com/somefrills/config/ModMenuIntegration.java
package com.somefrills.config;

import com.somefrills.misc.Utils;
import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import net.minecraft.client.gui.screen.Screen;

public class ModMenuIntegration implements ModMenuApi {

    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return ModMenuIntegration::getConfigScreen;
    }

    private static Screen getConfigScreen(Screen previous) {
        return Utils.getGuiScreen(previous);
    }
}

File: src/main/kotlin/com/somefrills/config/ModMenuIntegration.kt
package com.somefrills.config

import com.somefrills.utils.GuiUtils
import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import net.minecraft.client.gui.screens.Screen

class ModMenuIntegration : ModMenuApi {

    override fun getModConfigScreenFactory(): ConfigScreenFactory<Screen> {
        return ConfigScreenFactory<Screen> { previous ->
            GuiUtils.getGuiScreen(previous)
        }
    }
}